safe-global/safe-wallet-web
# feat: poll recovery state of Safe

## What it solves

Resolves #2752

## How this PR fixes it

This adds a new `recovery` slice to Redux. When a Safe is opened, and subsequently every ten minutes, the modules of it are parsed and for every Delay Modifier, details loaded.

```ts
type RecoveryQueueItem = TransactionAddedEvent & {
  timestamp: number
  validFrom: BigNumber
  expiresAt: BigNumber | null
}

type RecoveryState = Array<{
  address: string
  modules: Array<string>
  txExpiration: BigNumber
  txCooldown: BigNumber
  txNonce: BigNumber
  queueNonce: BigNumber
  queue: Array<QueuedTransactionAdded>
}>
```

## How to test it

Open a Safe with a Delay Modifier enabled, e.g. `gor:0xAecDFD3A19f777F0c03e6bf99AAfB59937d6467b` and observe the state load in Redux.

## Screenshots

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/0510b3eb-acea-4c01-8bcd-58d4717b4009)

## Checklist
